### DISCUSSION TITLE
One way binding component parameter to source

### DISCUSSION THREAD

**Discussion by danielchalmers** [dotnet/aspnetcore#48656] [Q&A] [answered]
Is there an equivalent to WPF's `OneWayToSource` binding mode so I can always have the result of my component's parameter in the parent?

**Comment by david-acker** [comment, marked as answer]
I'm not super familar with Blazor or WPF, so I'm not sure if this is 100% equivalent, but you could use an `EventCallback` to notify the parent component when the component parameter is changed in the child.

Example from the docs: [Binding with component parameters](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/data-binding?view=aspnetcore-7.0#binding-with-component-parameters)

```csharp
[Parameter]
public int Year { get; set; }

[Parameter]
public EventCallback<int> YearChanged { get; set; }

private async Task UpdateYearFromChild()
{
    await YearChanged.InvokeAsync(r.Next(1950, 2021));
}
```
